### DISCUSSION TITLE
Spectrum libraries with modifications

### DISCUSSION THREAD

**Discussion by SamuelLozanoJuarez** [vdemichev/DiaNN#1240] [Q&A] [answered]
Hi,
I am trying to use DIA-NN to analyse some samples and I have created a spectra library containing the minimum necessary fields: _ModifiedPeptide_, _PrecursorCharge_, _PrecursorMz_, _Tr_recalibred_, _ProductMz_ and _LibraryIntensity_. 
In this library there are several modified peptidoforms, with modifications indicated in the ModifiedPeptide field in square brackets (e.g. AAGA[-6.010321]ELVGMEDLADQLK). 
However, when I run DIA-NN, I get the following error message: 
`ERROR: D:\diannsrc\diann.cpp: 4064: unknown modification: -1.52499` (or any other value).
How should I include modifications in the library to avoid this error?

Thank you for your time

**Comment by vdemichev** [comment, marked as answer]
Hi Samuel,

You need to define each such modification using --mod. In this case whatever is between parentheses is treated by DIA-NN as a name, not as a number, i.e. can be just any sequence of characters. So the syntax will be: `--mod -6.010321,-6.010321`.

Best,
Vadim

**Reply by vdemichev** [reply]
If many of those, always possible to write an R or Python script to generate such definitions automatically.
